testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oauth/DPoPTest.java

```diff
@@ -419,7 +474,21 @@ public void testDPoPProofInvalidType() throws Exception {
 
     @Test
     public void testDPoPProofInvalidSignature() throws Exception {
-        testDPoPProofFailure(generateSignedDPoPProof(UUID.randomUUID().toString(), HttpMethod.POST, oauth.getEndpoints().getToken(), (long) Time.currentTime(), Algorithm.PS256, jwsEcHeader, rsaKeyPair.getPrivate(), null), "DPoP verification failure: org.keycloak.exceptions.TokenSignatureInvalidException: Invalid token signature");
+        DPoPGenerator customDPoPGenerator = new DPoPGenerator() {
```

Comment:
If this custom DPoPGenerator is the same as one in L445-458, it might be declared as an internal class to avoid duplication.

Reply:
Done, Thanks for the suggestion